libbpf-cargo/src/btf/btf.rs

```diff
@@ -731,7 +731,7 @@ impl<'a> Btf<'a> {
     fn load_type(&mut self, data: &'a [u8]) -> Result<BtfType<'a>> {
         let t = data.pread::<btf_type>(0)?;
         let extra = &data[size_of::<btf_type>()..];
-        let kind = (t.info >> 24) & 0xf;
```

Comment:
please check if it's the only place where we extract kind from btf_type. It's probably best to move this into a separate function, just like libbpf does.

Reply:
OK, will follow up on a new PR.

Currently, the code already has a `get_kind()` method, but it is used to extract kind_flag actually.